docs/tutorials/Upgrading_to_Application_Commands.md

```diff
@@ -0,0 +1,309 @@
+
+Regardless of the response, it should be picked based on the specific interaction the user is taking.
+
+#### Example
+
+For a `/search` command that searches an external service, an app could respond with an ephemeral message rather than posting back into the channel where users may not have context or want messages cluttering the channel.
+
+Ephemeral messages can also contain message components, so when relevant, there could be a button to share the information in the ephemeral message to a channel.
+
+![Example of ephemeral message to show search results](examples-ephemeral-message.png)
+
+## Onboarding users
+
+Once you’ve implemented slash commands (or really any feature for your app), it can be helpful to onboard existing and new users. While every app on Discord is unique, this section tries to generalize some good onboarding strategies, specifically to send updates to existing users.
+
+### Communicating app updates to users
+
+When you add new features or make major changes to your app, communicate what they are and how it specifically affects them. What do they need to do differently? What improves? What won’t they be able to do anymore?
+
+As to where to communicate the changes, you can start with any communication channels you own—a support server, blog, or social media account for example.
+
+You can also inform users about changes within the servers your app is installed in as long as it’s done in a non-intrusive way. If your app has a dedicated channel in posts, that would be a good place.
+
+> warn
+> Don’t DM all users in a server where your app is installed. It could get your app rate limited, but more importantly, it can be pretty intrusive and might lead to your app being uninstalled.
+
+#### Example
+
+The following is an example of an app update that may be sent to communicate the new way to access commands. Depending on the different features your app adopts (like options, subcommands, permissions, etc.), an update message or changelog entry will look very different.
+
+![Example message updating users about a new feature](examples-update-message.png)
+
+### Adding reactive education
+
+In addition to proactively communicating changes, you can reactively educate users when making changes to your app.
+
+For example, if your app currently replies to messages that start with `!poll`, you can continue listening to those messages for a couple of weeks and redirect users to a new `/poll` command using an ephemeral message.
```

Comment:
You can't respond to normal messages ephemerally

Reply:
rip sry will fix 🤦